src/zkvyper/arguments.rs

```diff
@@ -156,10 +156,14 @@ impl Arguments {
             }
         }
 
-        if self.version && std::env::args().count() > 2 {
-            anyhow::bail!(
-                "Error: No other options are allowed while getting the compiler version."
-            );
+        if self.version {
+            if std::env::args().count() > 2 {
+                anyhow::bail!(
+                    "Error: No other options are allowed while getting the compiler version."
```

Comment:
Just wondering why we have this condition in general.
Usually, for the most of command line utilities and compilers if `--version` specified, it prints the version regardless of other parameters, but never fails.

Reply:
Yeah, I've probably gone too far with my principle of *only used data allowed*.
Will reconsider this while working on the code coverage.